### DISCUSSION TITLE
Sync UI using kubectl

### DISCUSSION THREAD

**Discussion by aakashnand** [argoproj/argo-cd#7762] [Q&A]
Is there any way to sync argocd app on UI just by using `kubectl` command from CLI? For example, some of our developers prefer to use `kubectl apply` from the terminal. In such a case is it possible for ArgoCD to automatically check the app status and update its UI as `synced` without always waiting for the git code to be pushed?

**Comment by chetan-rns** [comment]
Hi @aakashnand, Argo CD has an annotation to refresh the app `argocd.argoproj.io/refresh` that could be applied using kubectl annotate cmd. But I don't think we have a similar annotation for sync. We could also use the argocd CLI and sync the app

**Reply by aakashnand** [reply]
@chetan-rns Thanks for your comment. What is the difference between refresh and sync? 

My actual problem:
I have an app created in ArgoCD which is synced from the git repo. 

When I make changes to the repo **on local** (not pushed to remote git yet) and I deploy those changes manually using `kubectl apply -f manifest/yml`. Now if I log in to ArgoCD it shows not in sync and it does not recognize that deployment has been changed because of local work. What I want ArgoCD to do is automatically detect the changes in deployment when I execute `kubectl apply` irrespective of my git push status. Is there any way to achieve this? 

Does argocd CLI work without code being pushed to git?

**Reply by chetan-rns** [reply]
@aakashnand Refresh will update the cache by pulling the latest changes and sync will reconcile the cluster state with these changes. We could use the `argocd app sync --local <path>` cmd to sync the app with changes made locally. No git queries will be made and Argo CD will use the resources present locally instead of fetching them from a git repo

**Reply by aakashnand** [reply]
Woah great I will try to use this once I confirm I will mark the solution. Let me try, I will update soon

**Reply by aakashnand** [reply]
@chetan-rns One question though, if my argoCD app is configured as auto-sync with git and I used `argocd app sync --local <path>` and then pushed my code to git, will it remove the existing deployment or it will just update using the existing deployment?

**Reply by chetan-rns** [reply]
@aakashnand I don't think Argo CD allows local sync when the automatic sync policy is enabled

```
FATA[0000] Cannot use local sync when Automatic Sync Policy is enabled except with --dry-run
```
